gateway-server/src/main/java/org/apache/knox/gateway/services/token/impl/DefaultTokenStateService.java

```diff
@@ -168,15 +167,15 @@ public boolean isExpired(final JWTToken token) {
 
   @Override
   public boolean isExpired(final String token) {
-    boolean isExpired;
+    boolean isUnknownToken;
 
-    isExpired = isRevoked(token); // Check if it has been revoked first
-    if (!isExpired) {
-      // If it has not been revoked, check its expiration
-      isExpired = (getTokenExpiration(token) <= System.currentTimeMillis());
+    isUnknownToken = isUnknown(token); // Check if the token exist
```

Comment:
Should this be using validateToken() for the unknown check?

Reply:
Yup, I was trying to avoid refactoring other parts, but this simplifies things.

Reply:
Actually, we cannot use validateToken(). validateToken() method throws an exception and does not return a boolean value. So calls to isExpired() will now start throwing exceptions. I could wrap validateToken() with try catch to get better reuse the code.

Reply:
Ok, just use isUnknown() as isRevoked() was used originally.

Reply:
Ok, reverted to older change.